includes/class-wc-gocardless-gateway.php

```diff
@@ -2315,11 +2315,58 @@ protected function _process_billing_request_event_cancelled( array $event ) {
 		if ( empty( $payment ) && empty( $mandate ) ) {
 			$order->update_status( 'cancelled', __( 'Billing request cancelled.', 'woocommerce-gateway-gocardless' ) );
 			$this->update_order_resource( $order, 'billing_request', $billing_request );
+			$this->handle_subscription_cancellation( $order, __( 'Billing request cancelled by customer.', 'woocommerce-gateway-gocardless' ) );
```

Comment:
Not sure if this is needed, as subscriptions will get cancelled with payment cancelled webhook call, and order is already cancelled on line no 2316. am I missing anything here? Thanks